src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -39071,6 +39071,200 @@ static bool G(out object o)
             comp.VerifyDiagnostics();
         }
 
+
+        [Fact, WorkItem(31370, "https://github.com/dotnet/roslyn/issues/31370")]
+        public void SuppressNullableWarning_Deconstruction()
+        {
+            var source = @"
+class C<T>
+{
+}
+class C2
+{
+    void M(C<string?> c)
+    {
+        // line 1
+        (string d1, (C<string> d2, string d3)) = (null, (c, null));
+
+        // line 2
+        (string e1, (C<string> e2, string e3)) = (null, (c, null))!;
+
+        // line 3
+        (string f1, (C<string> f2, string f3)) = (null, (c, null)!);
+
+        // line 4
+        (string g1, (C<string> g2, string g3)) = (null!, (c, null)!);
+
+        // line 5
+        (string h1, (C<string> h2, string h3)) = (null!, (c!, null!)); // no warnings
```

Comment:
>(null!, (c!, null!)); // no warnings [](start = 49, length = 36)

I had misunderstood yesterday. I thought this was producing undesired warnings...
It's nice that this suppression works.

Reply:
Got it. Thanks